includes/activity/class-event.php

```diff
@@ -0,0 +1,23 @@
+<?php
```

Comment:
why these classes? I tried to avoid them, because otherwise we also have to map all other types https://github.com/Automattic/wordpress-activitypub/blob/master/includes/transformer/class-post.php#L398

Reply:
We are still evaluating all pros and cons. At the moment we have only tried to get an idea of what the consequences would be. As is often the case: the deeper I dig into this repository, the more I can understand certain design decisions.

Reply:
True, but there are things that are simply legacy decisions! Be aware that this project started as a spare time thing and still has some crappy code everywhere. So do not try to understand everything and feel free to challenge us, that's what I also try here ;)

I discuss to understand, not to win :)

Reply:
Thank you, for offering help, even at our _Drafting Stage_. I gladly accept this invitation for discussions :)

One of my goals is,  that it is very easy an straight forward, for feature developers adding their own transformers to their plugins. Very likely they will have to extend the _Base_Object_ class anyway, especially if they want to add any custom fields that are not defined yet in our _Base_Objet_ class, which until now is consistent with the _Activitystreams Vocabulary Standard_. For example even for supported object types like 'Event' Mobilizon makes usage of PeerTubes `commentsEnabled` feature.

Other developers might have a look, how thinks are done in this project, and might get confused if it does things a bit differently than the future docs advice them to do.

Another related issue is, that when adding a custom field, one also has to add the appropriate json-ld context. 
Though the filter `activitypub_json_context` is available, editing the array directly seemed kind of rough to me and applying the filter within the transformer code leaded to a runtime issue, so I had to apply this filter in a different part of the code, which is not ideal in my opinion.

I will share our repository with some example transformers registering to the ActivityPub-plugin hopefully today, after doing some renaming and initial documentation.

Reply:
> I will share our repository with some example transformers registering to the ActivityPub-plugin hopefully today, after doing some renaming and initial documentation.

That would be perfect!

I agree with your point, but I am not sure if the "extensions" (Event-Activity) should be part of this repo or part of yours. I agree, that it has to be as easy as possible and maybe we have to change the current context implementation for that.

You can maybe have a look what I did to extend the base object(s) for the Follower (https://github.com/Automattic/wordpress-activitypub/blob/master/includes/model/class-follower.php) or internal User mapping (https://github.com/Automattic/wordpress-activitypub/blob/master/includes/model/class-user.php).